src/basic-languages/mdx/mdx.test.ts

```diff
@@ -0,0 +1,8 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { testTokenization } from '../test/testRunner';
+
+testTokenization('mdx', []);
```

Comment:
Please provide some tests here, so we can refactor the code base without having to worry about unknowingly breaking external contributions!

Reply:
I would love to! But tests don’t appear to work for me. Following the [contribution guide](https://github.com/microsoft/monaco-editor/blob/main/CONTRIBUTING.md#running-the-editor-tests):

```sh
$ npm install

> monaco-editor@0.33.0 postinstall
> ts-node ./build/postinstall

Installing husky hooks...
$ husky install
husky - Git hooks installed

added 254 packages, and audited 255 packages in 4s

found 0 vulnerabilities
$ npm run release

> monaco-editor@0.33.0 release
> ts-node ./build/build && ts-node ./build/release

Launching compiler at src/tsconfig.json...
Compiled src/tsconfig.json
Ran prettier over out/release/css.d.ts
Ran prettier over out/release/html.d.ts
Ran prettier over out/release/json.d.ts
Ran prettier over out/release/typescript.d.ts
Copied src/language/typescript/lib/typescriptServices-amd.js to out/amd/language/typescript/lib/typescriptServices.js
ADDING ThirdPartyNotices from ./ThirdPartyNotices.txt
$ npm test       

> monaco-editor@0.33.0 test
> mocha test/unit/all.js

TypeError: nls.getConfiguredDefaultLocale is not a function
    at Object.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/monaco-editor-core/dev/vs/editor/editor.main.js:38449:38)
    at Object.execCb (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1946:33)
    at Module.check (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1128:55)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1426:22)
    at Object.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1807:39)
    at Module.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1411:33)
    at /home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:384:23
    at each (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:309:31)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1363:17)
    at Object.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1807:39)
    at Module.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1411:33)
    at /home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:384:23
    at each (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:309:31)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1363:17)
    at Object.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1807:39)
    at Module.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1411:33)
    at /home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:384:23
    at each (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:309:31)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1363:17)
    at Module.init (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1038:26)
    at callGetModule (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1453:63)
    at Object.completeLoad (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1840:21)
    at req.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:2613:17)
    at Object.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1935:21)
    at Module.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1084:29)
    at Module.fetch (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1074:66)
    at Module.check (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1106:30)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1426:22)
    at Object.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1807:39)
    at Module.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1411:33)
    at /home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:384:23
    at each (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:309:31)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1363:17)
    at Module.init (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1038:26)
    at callGetModule (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1453:63)
    at Object.completeLoad (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1840:21)
    at req.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:2613:17)
    at Object.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1935:21)
    at Module.load (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1084:29)
    at Module.fetch (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1074:66)
    at Module.check (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1106:30)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1426:22)
    at Object.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1807:39)
    at Module.<anonymous> (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1411:33)
    at /home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:384:23
    at each (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:309:31)
    at Module.enable (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1363:17)
    at Module.init (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1038:26)
    at Timeout._onTimeout (/home/remco/Projects/monaco-editor/node_modules/requirejs/bin/r.js:1710:36)
    at listOnTimeout (node:internal/timers:564:17)
    at process.processTimers (node:internal/timers:507:7) {
  requireMap: {
    prefix: undefined,
    name: 'vs/base/common/platform',
    parentMap: null,
    unnormalized: false,
    url: './node_modules/monaco-editor-core/dev/vs/base/common/platform.js',
    originalName: 'vs/base/common/platform',
    isDefine: true,
    id: 'vs/base/common/platform'
  },
  requireModules: [ 'vs/base/common/platform' ],
  requireType: 'define'
}
```

This is all run on the main branch up-to-date with the `microsoft/monaco-editor` repository. I also looked at the CI workflow, as that’s passing, but I don’t see what I’m doing wrong.

Reply:
I followed those instructions, but it’s not working for me.

To summarize my steps:

```sh
npm install     # This is fine
npm run release # This is fine too
npm test        # This fails
```

Reply:
you need vscode nls next to the monaco editor repo.

Reply:
~~I now have the following sibling folders:~~

- ~~`monaco-editor`~~
- ~~`vscode`~~
- ~~`vscode-loc`~~
- ~~`vscode-nls`~~

~~But I still get the same error. Also it really consuses me that the [CI workflow](https://github.com/microsoft/monaco-editor/blob/main/.github/workflows/ci.yml) doesn’t clone anything~~

Edit: I see what’s going on now. CI is broken. [The tests **are** broken](https://github.com/microsoft/monaco-editor/runs/7468963595#step:11:9), but the tests incorrectly exit with a succesful status code.

Reply:
I think both these lines need to contain `process.exitCode = 1;`. I’m not going to make a PR for it, because I don’t know how the fix the underlying issue related to `nls`.

https://github.com/microsoft/monaco-editor/blob/9d278685b078158491964f8fd7ac9628fffa0f30/test/unit/all.js#L52-L61

Reply:
That should be fixed now.

Reply:
I'm going to approve this PR, but please provide tests in a follow up PR - otherwise we cannot guarantee that mdx support breaks at some point.